crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -468,6 +468,45 @@ impl Backend {
         }
     }
 
+    /// Reset to fresh state
+    pub async fn reset_to_non_fork(&self) -> Result<(), BlockchainError> {
+        // Reset fork
+        *self.fork.write() = None;
+
+        // Clear env
+        {
+            let mut env = self.env.write();
+            env.clear();
+
+            // Set default anvil chain_id
+            env.cfg.chain_id = DEV_CHAIN_ID;
+
+            // Reset time
+            self.time.reset(env.block.timestamp.to::<u64>());
+        }
```

Comment:
this isn't entirely accurate, what we need to consider here is that the node can be in two modes here, forked or mem only.

if mem only we need to reset to the genesis block

if in forked, we need to entirely reinitialize the node block env like this

Reply:
@mattsse So if the node is in mem mode, we just clear the env like `env.clear()`?